### PR TITLE
fix(j2p): support sint/fixed/sfixed types convertion and generic call

### PR DESCRIPTION
**Pull request by khan-yin**
#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [ ] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io)

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en: support sint/fixed/sfixed types convertion and generic call
zh(optional): 支持 sint/fixed/sfixed 基本类型的json协议转换和泛化调用

### CODE REVIEW

File: conv/j2p/decode.go [hunk 1]
```diff
@@ -260,12 +272,26 @@ func (self *visitorUserNode) OnInt64(v int64, n json.Number) error {
...
 	switch fieldDesc.Kind() {
-	case proto.Int32Kind, proto.Sint32Kind, proto.Sfixed32Kind, proto.Fixed32Kind:
+	case proto.Int32Kind:
 		convertData := *(*int32)(unsafe.Pointer(&v))
-
 		if err = self.p.WriteInt32(convertData); err != nil {
 			return err
 		}
+	case proto.Sint32Kind:
+		convertData := *(*int32)(unsafe.Pointer(&v))
```
Comment (AsterDY) [inline conv/j2p/decode.go:281]: 这里为啥要用unsafe?这样转出来的值不对吧。应该用 int32(v) 这样

File: proto/binary/binary.go [hunk 2]
```diff
@@ -240,23 +239,25 @@ func (p *BinaryProtocol) WriteUint32(value uint32) error {
...
 
 // WriteSfixed32
 func (p *BinaryProtocol) WriteSfixed32(value int32) error {
-	v, err := p.malloc(4)
-	if err != nil {
-		return err
-	}
-	binary.LittleEndian.PutUint32(v, uint32(value))
-	return err
+	// v, err := p.malloc(4)
```
Comment (AsterDY) [inline proto/binary/binary.go:254]: 不需要的代码就删掉吧

File: testdata/kitex_gen/pb/base/base.pb.go
Comment (AsterDY) [inline testdata/kitex_gen/pb/base/base.pb.go:1]: 这些service文件测试应该用不到，拿掉吧

**Review by AsterDY — APPROVED** [review]

File: conv/j2p/decode.go [hunk 3]
```diff
@@ -185,18 +185,30 @@ func (self *visitorUserNode) OnBool(v bool) error {
 		self.inskip = false
 		return nil
 	}
+
 	var err error
-	if self.globalFieldDesc == nil {
-		return newError(meta.ErrConvert, "self.globalFieldDescriptor is nil, type Onbool", nil)
+	top := self.stk[self.sp]
+	fieldDesc := self.globalFieldDesc
```
Comment (AsterDY) [inline conv/j2p/decode.go:191]: 这个是修复list的一些bug？

Comment (khan-yin) [reply conv/j2p/decode.go:191]: 是的，主要是兼容了一下repeated bool字段

**Review by AsterDY — APPROVED** [review]

**Pull request merged**